analytics/management/commands/create_ga_report.py

```diff
@@ -0,0 +1,325 @@
+                                                                                   start_date, end_date,
+                                                                                   dimensions, metrics,
+                                                                                   other_dimensions, other_filters)
+
+        # Merge the statistics from the different data sources and print them
+        self._print_statistics(db_statistics_by_page_url, ga_statistics, other_ga_stats)
+
+    def valid_date(self, date_string):
+        try:
+            datetime.strptime(date_string, "%Y-%m-%d")
+        except ValueError:
+            message = "Not a valid date: '{0}'.".format(date_string)
+            raise argparse.ArgumentTypeError(message)
+        else:
+            return date_string
+
+    def _print_statistics(self, db_statistics_by_page_url, ga_statistics_by_page_url, other_ga_stats_by_page_url):
+        header_row = ['Title', 'URL', 'Published Date', 'Days Published', 'Parent Page', 'Category', 'Type',
+                      'Word Count', 'Pageviews (PV)', 'Unique Pageviews (UPV)', 'Avg. Time on Page (in seconds)',
+                      'facebook PV', 'facebook UPV', 'twitter PV', 'twitter UPV', 'direct PV', 'direct UPV',
+                      'google PV', 'google UPV']
+        print '\t'.join([x for x in header_row])
+        for url in db_statistics_by_page_url.keys():
+            page_data = db_statistics_by_page_url[url]
+            data_row = self._format_page_data_for_output(page_data)
+            ga_statistics = ga_statistics_by_page_url[url]
+            assert len(ga_statistics) == 1
+            other_statistics = other_ga_stats_by_page_url[url]
+            data_row.extend(self._format_ga_data_for_output(ga_statistics[0], other_statistics))
+            print '\t'.join([x for x in data_row])
+
+    def _format_page_data_for_output(self, page_data):
+        output = []
+        output.append(page_data['title'])
+        output.append(page_data['url'])
+        published_date = page_data['published_date'].strftime('%Y-%m-%d')
+        output.append(published_date)
+        output.append(unicode(page_data['days_published']))
+        output.append(page_data['parent_page'])
+        output.append(page_data['category'])
+        types = []
+        if page_data['is_video']:
+            types.append('Video')
+        if page_data['is_interview']:
+            types.append('Interview')
+        if page_data['is_visualization']:
+            types.append('Visualization')
+        output.append(','.join(types))
+        word_count = sum([x for x in page_data['word_count_by_type'].values()])
+        output.append(unicode(word_count))
+        return output
+
+    def _format_ga_data_for_output(self, ga_statistics, other_statistics):
+        # TODO: Here is the perfect example of why this Command isn't generic enough...
+        output = [unicode(x) for x in ga_statistics]
```

Comment:
Should you be decoding from a known character encoding, say UTF-8 to get a unicode string at this point?